### ISSUE TITLE
Replace pytz with zoneinfo

### ISSUE THREAD

**Issue by hackdna** [onnela-lab/forest#208]
The [zoneinfo](https://docs.python.org/3/library/zoneinfo.html) module provides a concrete time zone implementation to support the IANA time zone database as originally specified in [PEP 615](https://peps.python.org/pep-0615/).

Note that this requires upgrading Python to 3.9+ or using `backports.zoneinfo`: https://pypi.org/project/backports.zoneinfo/

Additional info:
* [Paul Ganssle: Working with Time Zones: Everything You Wish You Didn't Need to Know](https://www.youtube.com/watch?v=rYzgroaK8_Q)
* migrating from pytz to zoneinfo: https://pypi.org/project/pytz-deprecation-shim/

**Comment by biblicabeebli** [comment]
I use dateutil over in beiwe-backend, dropped pytz due to bugs and haven't had an issue with it at all.

**Comment by biblicabeebli** [comment]
Actually you need to stay on 3.8 for now. Python version compatability needs to be the same as the beiwe backend.

3.8 is about a year out from end of life so I will be upgrading, probably to 3.11, for the beiwe backend's server stacks in the next year.

I need to update the Celery backend and server platforms substantially for the task queue/dispatch, and will take a look at divorcing the python version for forest so you don't have this constraint anymore and so users have more control when dispatching forest tasks from the beiwe website. Current environment is Ubuntu 18.04 running Python 3.8.x installed via Pyenv.

**Comment by hackdna** [comment]
Yes, I am aware we have to stick with 3.8. It would be great to use a more recent version of Python to be able to take advantage of the new features and performance improvements.

Is Celery a blocker for Python upgrade? Also, any particular reason to still use Ubuntu 18.04? If we could use a recent version of Python for both Beiwe and Forest then no need to install two separate versions.

**Comment by biblicabeebli** [comment]
Celery's not the blocker, its just out of date and the upgrade to celery 5 is more than drop-in.

"The blocker" is that I have to ... update the AWS platform, document/directions for anyone running 3.8 to upgrade in-place to 3.11 on elastic beanstalk, upgrade the spinup from zero script, rev 18.04 -> 22.04 (unless 24.04 is out and an LTS by then?) along with concurrently swapping RabbitMQ (causes headaches and configuration management changed substantially after 18.04) to Redis _and upgrade Celery_ because everything listed is related to it.

🫠

Technical debt.

**Comment by biblicabeebli** [comment]
This work has been merged into the ruff-check-reflow-and-typing branch, which is ahead of develop as it will require a complex merge once outstanding work is complete.
